### PR TITLE
Roadhog ultimate fix

### PR DESCRIPTION
**Pull request by snappycreeper**
fixes my previous pr #243, and changes health and sets facing back to old one. thanks to @MrKingMichael for help with the facing position

### CODE REVIEW

**Comment by MrKingMichael** [comment]
It works, Roadhog does not gain invulnerability after being respawned.

Some nitpicks:

1. Hack should immediately cancel.
2. This waits until the status effect has worn off, but it doesn't check if you're still using the ultimate, leading to scenarios were this executes for no reason. To replicate this: Start using the ultimate then after a few seconds get hacked by Sombra. The ultimate will wear off, and when the hack status wears off as well, you'll be respawned.
3. It does take into account the player's direction, meaning that your character in most scenarios will point somewhere else after respawning.
4. This should be a function, considering other heroes might need to make use of this. It could be used to set Roadhog's ult duration to OW1's duration, for example.

**Comment by MaxwellJung** [comment]
> It works, Roadhog does not gain invulnerability after being respawned.
> 
> Some nitpicks:
> 
> 1. Hack should immediately cancel.
> 2. This waits until the status effect has worn off, but it doesn't check if you're still using the ultimate, leading to scenarios were this executes for no reason. To replicate this: Start using the ultimate then after a few seconds get hacked by Sombra. The ultimate will wear off, and when the hack status wears off as well, you'll be respawned.
> 3. It does not take into account the player's direction, meaning that your character in most scenarios will face somewhere else after respawning.
> 4. This should be a function, considering other heroes might need to make use of this. It could be used to set Roadhog's ult duration to OW1's duration, for example.

Are these issues resolved? @snappycreeper @MrKingMichael?

**Comment by snappycreeper** [comment]
Yes, all of them except number 4 because I didn't make a function for it. But everything else works perfectly

**Comment by MaxwellJung** [comment]
> Yes, all of them except number 4 because I didn't make a function for it. But everything else works perfectly

Yeah I might abstract this further into a utility function that checks if you're cc'd so we don't have to type the long boolean logic every time.

File: src/ow1/heroes/roadhog.opy [hunk 1]
```diff
@@ -67,3 +70,32 @@ rule "[roadhog.opy]: Stop autofire when not in ult":
...
+    if eventPlayer.isUsingUltimate() == true:
+        eventPlayer.roadhog_ult_pos = eventPlayer.getPosition()
+        eventPlayer.roadhog_ult_health = eventPlayer.getHealth()
+        kill(eventPlayer,null)
+        eventPlayer.respawn()
+        eventPlayer.setHealth(eventPlayer.roadhog_ult_health)
+        eventPlayer.setFacing(eventPlayer.roadhog_ult_facing, Relativity.TO_WORLD)
+        eventPlayer.startForcingPosition(eventPlayer.roadhog_ult_pos, true)
+        wait(0.3)
+        eventPlayer.stopForcingPosition()
```
[the remark this answers is not in the document]
Comment (snappycreeper) [reply src/ow1/heroes/roadhog.opy]: I didn't know that was a function, I replaced it with teleport now.

File: src/ow1/heroes/roadhog.opy [hunk 2]
```diff
@@ -67,3 +70,32 @@ rule "[roadhog.opy]: Stop autofire when not in ult":
...
+    
+    if eventPlayer.isUsingUltimate() == true:
+        eventPlayer.roadhog_ult_pos = eventPlayer.getPosition()
+        eventPlayer.roadhog_ult_health = eventPlayer.getHealth()
+        kill(eventPlayer,null)
+        eventPlayer.respawn()
+        eventPlayer.setHealth(eventPlayer.roadhog_ult_health)
+        eventPlayer.setFacing(eventPlayer.roadhog_ult_facing, Relativity.TO_WORLD)
+        eventPlayer.startForcingPosition(eventPlayer.roadhog_ult_pos, true)
+        wait(0.3)
```
[the remark this answers is not in the document]
Comment (snappycreeper) [reply src/ow1/heroes/roadhog.opy]: I was just using it to stop forcing the position. I removed it in latest commit.

**Comment by MrKingMichael** [comment]
Works great now. I'd only suggest abstracting it like @MaxwellJung suggested and turning this into a function, maybe Doomfist could make use of this after using his ult to remove the empowered punch.

**Review by MaxwellJung — APPROVED** [review]

**Pull request merged**